appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy

```diff
@@ -12,168 +12,257 @@ import java.time.ZoneId
+		assert dayActivityForTimeZoneGoal.spread.size() == 96
 
 		def inactivityDayActivityForBudgetGoal = dayActivityOverview._embedded."yona:dayActivities".find{ it._links."yona:goal".href == budgetGoalNewsUrl}
-		inactivityDayActivityForBudgetGoal.totalActivityDurationMinutes == 0
-		inactivityDayActivityForBudgetGoal.totalMinutesBeyondGoal == 0
+		assert inactivityDayActivityForBudgetGoal.totalActivityDurationMinutes == 0
+		assert inactivityDayActivityForBudgetGoal.totalMinutesBeyondGoal == 0
 	}
 
 	def 'Get day activity detail'()
 	{
 		given:
-		def richard = addRichard()
-		def goals = appService.getGoals(richard)
-		def budgetGoalGamblingUrl = findGoal(goals, GAMBLING_ACT_CAT_URL)._links.self.href
-		def budgetGoalNewsUrl = findGoal(goals, NEWS_ACT_CAT_URL)._links.self.href
-		analysisService.postToAnalysisEngine(richard, ["Gambling"], "http://www.poker.com")
+		setupTestScenario()
 
 		when:
-		def overviewsResponse = appService.getDayActivityOverviews(richard)
-		overviewsResponse.responseData._embedded."yona:dayActivityOverviews"
-		overviewsResponse.responseData._embedded."yona:dayActivityOverviews".size() == 1
+		true
+
+		then:
+		testDayActivityDetail(bob, bob.dailyActivityReportsUrl, bobBudgetGoalGamblingUrl, bobBudgetGoalNewsUrl)
+	}
+
+	def 'Get day activity detail from buddy'()
+	{
+		given:
+		setupTestScenario()
+
+		when:
+		true
+
+		then:
+		testDayActivityDetail(richard, bobDailyActivityReportsUrlForRichard, bobBudgetGoalGamblingUrlForRichard, bobBudgetGoalNewsUrlForRichard)
+	}
+
+	void testDayActivityDetail(fromUser, dailyActivityReportsUrl, budgetGoalGamblingUrl, budgetGoalNewsUrl)
+	{
+		def overviewsResponse = appService.getResourceWithPassword(dailyActivityReportsUrl, fromUser.password)
+		assert overviewsResponse.responseData._embedded."yona:dayActivityOverviews"
+		assert overviewsResponse.responseData._embedded."yona:dayActivityOverviews".size() == 1
 		def dayActivityOverview = overviewsResponse.responseData._embedded."yona:dayActivityOverviews"[0]
-		dayActivityOverview._embedded."yona:dayActivities"
-		dayActivityOverview._embedded."yona:dayActivities".size() == 2
+		assert dayActivityOverview._embedded."yona:dayActivities"
+		assert dayActivityOverview._embedded."yona:dayActivities".size() == 3
 		def dayActivityForGoal = dayActivityOverview._embedded."yona:dayActivities".find{ it._links."yona:goal".href == budgetGoalGamblingUrl}
-		dayActivityForGoal?._links?.self?.href
-		def response = appService.getResourceWithPassword(dayActivityForGoal._links.self.href, richard.password)
+		assert dayActivityForGoal?._links?.self?.href
+		def response = appService.getResourceWithPassword(dayActivityForGoal._links.self.href, fromUser.password)
+		assert response.status == 200
+		assert response.responseData.spread
+		assert response.responseData.spread.size() == 96
+		assert response.responseData.totalActivityDurationMinutes == 1
+		assert response.responseData.goalAccomplished == false
```

Comment:
No, this test does not assert something about the time zone goal. It only asserts for `budgetGoalGamblingUrl`.